### PR TITLE
Allow users fill in a report form and see a chart representing requested data

### PR DESCRIPTION
**Pull request by alahab2**
This change creates a report form which can be accessed by clicking on "Reports" link in the `navbar`. After a User fills in the report form, a chart is generated to visualize the data.

**Note:** This change edits `donation.rb` seed file. Please run `rake db:drop db:create db:migrate db:seed`.

See the screenshots below:

---
Default view:
screencapture-localhost-3000-donations-reports-1479190379752

Donations for chosen year:
screencapture-localhost-3000-donations-reports-1479190436254

Donations for chosen year and month:
screencapture-localhost-3000-donations-reports-1479190458201

Donations for chosen year and cause:
screencapture-localhost-3000-donations-reports-1479190496446

Donations for chosen year, month and cause:
screencapture-localhost-3000-donations-reports-1479190585222

Donations for chosen year and event:
screencapture-localhost-3000-donations-reports-1479190519437

Donations for chosen year, month and event:
screencapture-localhost-3000-donations-reports-1479190665814

**Before submitting, check that:**

 - [ ] You have added (passing) tests for your code.
 - [x] You have written good* commit messages.
 - [ ] You have squashed relevant commits together.
 - [x] You have ensured that RuboCop is passing.
 - [x] Your PR relates to one subject, with a clear title and
       description using complete, grammatically correct sentences.

---

If your change contains views, ensure that:

 - [x] You have included a screenshot of the new view.

---

*If you do not know what makes a good commit message, or why it
is important, please refer to these resources: [1][1], [2][2], [3][3].

[1]: https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message
[2]: http://chris.beams.io/posts/git-commit/
[3]: https://github.com/blog/1943-how-to-write-the-perfect-pull-request

### CODE REVIEW

File: app/models/donation.rb [hunk 1]
```diff
@@ -9,5 +9,7 @@ class Donation < ActiveRecord::Base
 
   scope :search, ->(keyword) { joins(:donor).where("name ILIKE ?", "%#{keyword}%") }
 
-  scope :donations_per_year, ->(year) { where("EXTRACT(YEAR FROM created_at) = ?", year) }
+  scope :donations_per_year, ->(year) { where("EXTRACT(YEAR FROM donations.created_at) = ?", year) }
```
Comment (Drenmi) [inline app/models/donation.rb]: I would name this `.donations_for_year`, as "per year" might imply that we're getting results for several years.

File: app/models/report.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Report
+  attr_reader(:year, :month, :cause, :event)
```
Comment (Drenmi) [inline app/models/report.rb]: Method calls that are part of DSLs, or have "keyword status" in Ruby shouldn't use parentheses. 😀

File: app/models/report.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(params)
+    @year = params[:date][:year]
+    @month = params[:date][:month]
+    @cause = params[:cause_id]
```
Comment (Drenmi) [inline app/models/report.rb]: We shouldn't name these `cause` and `event` if they don't hold causes or events. I think what we mean to do here is to actually find these records in the initializer?

File: app/views/reports/_report_chart.html.slim [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.row.col-xs-12.col-md-10.offset-md-1.text-xs-center
+  .col-xs-6.col-md-3
+    <b>Year:</b> #{report.year}
+  .col-xs-6.col-md-3
+    <b>Month:</b> #{!report.month.empty? ? report.month : "All"}
```
Comment (Drenmi) [inline app/views/reports/_report_chart.html.slim]: This can be written: `report.month || "All"`

Comment (alahab2) [inline app/views/reports/_report_chart.html.slim]: When I do `report.month || "All"` it shows month's number if a month is chosen, but if not, it just shows blank :
screen shot 2016-11-15 at 2 38 30 pm

Comment (alahab2) [inline app/views/reports/_report_chart.html.slim]: Addressed all the comments except for this one.

Comment (Drenmi) [inline app/views/reports/_report_chart.html.slim]: Try `report.month.presence` || "All"`

:grinning:

**Comment by Drenmi** [comment]
Squash commits and 🚀 .

**Pull request merged**